contrib/storage-jdbc/src/main/java/org/apache/drill/exec/store/jdbc/JdbcRecordReader.java

```diff
@@ -180,22 +182,34 @@ private static String nameFromType(int javaSqlType) {
   }
 
   @Override
-  public void setup(OperatorContext operatorContext, OutputMutator output) throws ExecutionSetupException {
+  public void setup(OperatorContext operatorContext, OutputMutator output) {
     try {
       connection = source.getConnection();
       statement = connection.createStatement();
       resultSet = statement.executeQuery(sql);
 
-      final ResultSetMetaData meta = resultSet.getMetaData();
-      final int columns = meta.getColumnCount();
+      ResultSetMetaData meta = resultSet.getMetaData();
+      int columnsCount = meta.getColumnCount();
+      if (columns.size() != columnsCount) {
+        throw UserException
+            .validationError()
+            .message(
+              "Expected columns count differs from the returned one.\n" +
+                  "Expected columns: %s\n" +
+                  "Returned columns count: %s",
+              columns, columnsCount)
+            .addContext("sql", sql)
+            .addContext("plugin", storagePluginName)
+            .build(logger);
+      }
       ImmutableList.Builder<ValueVector> vectorBuilder = ImmutableList.builder();
       ImmutableList.Builder<Copier<?>> copierBuilder = ImmutableList.builder();
 
-      for (int i = 1; i <= columns; i++) {
-        final String name = meta.getColumnLabel(i);
-        final int jdbcType = meta.getColumnType(i);
-        final int width = meta.getPrecision(i);
-        final int scale = meta.getScale(i);
+      for (int i = 1; i <= columnsCount; i++) {
+        String name = columns.get(i - 1);
+        int jdbcType = meta.getColumnType(i);
```

Comment:
Please put the comment here that column index in `ResultSetMetaData` starts from 1

Reply:
Thanks, done.